pkg/controller/azdrivernode.go

```diff
@@ -101,12 +101,21 @@ func (r *ReconcileAzDriverNode) Recover(ctx context.Context) error {
 	}
 
 	for _, node := range nodes.Items {
-		updated := node.DeepCopy()
-		updated.Annotations = azureutils.AddToMap(updated.Annotations, consts.RecoverAnnotation, "azDriverNode")
-		if _, err = r.azClient.DiskV1beta2().AzDriverNodes(r.objectNamespace).Update(ctx, updated, metav1.UpdateOptions{}); err != nil {
-			return err
+		// if the corresponding node has been deleted, delete the azdriverode object
+		_, err := r.kubeClient.CoreV1().Nodes().Get(ctx, node.Spec.NodeName, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			azN := r.azClient.DiskV1beta2().AzDriverNodes(r.objectNamespace)
+			if err = azN.Delete(ctx, node.Name, metav1.DeleteOptions{}); err != nil {
+				return nil
```

Comment:
If delete current node is failed, it `return nil` immediately here in the for-loop. What about other nodes in the loop? I think we can continue to recover other nodes.

Reply:
We should log the error and continue rather than return.